openedx_lti_tool_plugin/models.py

```diff
@@ -307,3 +306,16 @@ def update_score(
     def __str__(self) -> str:
         """Get a string representation of this model instance."""
         return f'<LtiGradedResource, ID: {self.id}>'
+
+    def save(self, *args: tuple, **kwargs: dict):
+        """Model save method.
+
+        In this method we try to validate if the given context key is a valid
+        UsageKey or CourseKey.
+
+        Args:
+            *args: Variable length argument list.
+            **kwargs: Arbitrary keyword arguments.
+        """
+        super().save(*args, **kwargs)
+        validate_context_key(self.context_key)
```

Comment:
Did you try this?. @alexjmpb 
```suggestion
        self.full_clean()
        super().save(*args, **kwargs)
```